quiz/serializers.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import serializers
+from quiz.models import Question, Answer, Quiz
+from accounts.models import CustomUser as User
+
+
+class QuizSerializer(serializers.ModelSerializer):
+
+    
+    
+    class Meta:
+        model = Quiz
+        fields = [
+            'url',
+            'name', 
+            'category',
+            
+        ]
+
+
+class AnswerSerializer(serializers.ModelSerializer):
+
+    
+   
+    
+    class Meta:
+        model = Answer
+        fields = [
+            'id',
+            'text', 
+            'is_correct',
+            
+        ]
+
+
+class QuestionSerializer(serializers.ModelSerializer):
+
+    quiz = QuizSerializer(read_only=True)
+    answer = AnswerSerializer(read_only=True, many=True)
+
+    class Meta:
+    
+        model = Question
+        fields = [
+            'quiz',
+            'id',
+            'text',
+            'answer',       
+        ]
+
+    def increase_score(self):
+        user_score = User.score
+        if user_score >= 0:
+            user_score = user_score + 1
+        return user_score
+    
+    def decrease_score(self):
+        user_score = User.score
+        if user_score >= 0:
+            user_score = user_score - 1
```

Comment:
Se o score atual for igual a 0 e você diminuir 1 vai ficar -1. A regra diz que deveria ser no mínimo 0. Essa validação deveria ser apenas `>` e não `>=`.